hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/schemas/V057AddressBookSchema.java

```diff
@@ -47,29 +38,18 @@ public V057AddressBookSchema() {
     }
 
     @Override
-    public void restart(@NonNull final MigrationContext ctx) {
+    public void migrate(@NonNull final MigrationContext ctx) {
         requireNonNull(ctx);
-        final AtomicReference<Network> network = new AtomicReference<>();
-        if (ctx.isGenesis()) {
-            try {
-                network.set(ctx.startupNetworks().genesisNetworkOrThrow());
-            } catch (Exception ignore) {
-                // FUTURE - fail hard here once the roster lifecycle is always enabled
-                return;
-            }
-        } else {
-            ctx.startupNetworks().overrideNetworkFor(ctx.roundNumber()).ifPresent(network::set);
+        if (!ctx.configuration().getConfigData(AddressBookConfig.class).useRosterLifecycle()) {
+            return;
         }
-        if (network.get() != null) {
-            setNodeMetadata(network.get(), ctx.newStates());
+        if (ctx.isGenesis()) {
+            setNodeMetadata(ctx.startupNetworks().genesisNetworkOrThrow(), ctx.newStates());
         }
     }
 
-    private void setNodeMetadata(@NonNull final Network network, @NonNull final WritableStates writableStates) {
-        final WritableKVState<EntityNumber, Node> nodes = writableStates.get(NODES_KEY);
-        network.nodeMetadata().stream()
-                .filter(NodeMetadata::hasNode)
-                .map(NodeMetadata::nodeOrThrow)
-                .forEach(node -> nodes.put(new EntityNumber(node.nodeId()), node));
+    @Override
+    public void restart(@NonNull final MigrationContext ctx) {
+        AddressBookTransplantSchema.super.restart(ctx);
```

Comment:
Hmm, I think it shouldn't be required, will update it.

Reply:
Update: `.super` is needed otherwise compilation fails with `error: non-static method restart(MigrationContext) cannot be referenced from a static context` because we're calling the default method of the   `AddressBookTransplantSchema` interface.